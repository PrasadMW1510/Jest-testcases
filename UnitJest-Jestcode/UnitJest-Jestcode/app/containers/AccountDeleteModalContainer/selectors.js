import { createSelector } from 'reselect';

/**
 * Direct selector to the accountDeleteModalContainer state domain
 */
const selectAccountDeleteModalContainerDomain = state => state.get('accountDeleteModalContainer');

/**
 * Default selector used by AccountDeleteModalContainer
 */

const makeSelectAccountDeleteModalContainer = () =>
  createSelector(selectAccountDeleteModalContainerDomain, substate => substate.toJS());

export default makeSelectAccountDeleteModalContainer;
export { selectAccountDeleteModalContainerDomain };
