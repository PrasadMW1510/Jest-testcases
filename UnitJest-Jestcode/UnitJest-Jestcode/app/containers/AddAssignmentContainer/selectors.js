import { createSelector } from 'reselect';

/**
 * Direct selector to the addAssignmentContainer state domain
 */
const selectAddAssignmentContainerDomain = state => state.get('addAssignmentContainer');

/**
 * Other specific selectors
 */

/**
 * Default selector used by AddAssignmentContainer
 */

const makeSelectAddAssignmentContainer = () =>
  createSelector(selectAddAssignmentContainerDomain, substate => substate.toJS());

export default makeSelectAddAssignmentContainer;
export { selectAddAssignmentContainerDomain };
