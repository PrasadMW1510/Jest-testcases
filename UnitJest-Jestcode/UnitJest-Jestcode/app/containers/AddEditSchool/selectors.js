import { fromJS } from 'immutable';
import { createSelector } from 'reselect';
import { formValueSelector } from 'redux-form/immutable';
import { FORM_SCHOOL_PROFILE, META_DATA_GRADES } from './constants';

const selectForm = state => state.getIn(['form', FORM_SCHOOL_PROFILE]);

// TODO: See if the following definition can supplant the 'selectForms' definition above,
// the following selectors adjusting accordingly.
const selectSchoolForm = formValueSelector(FORM_SCHOOL_PROFILE);

/**
 * Direct selector to the addSchoolContainer state domain
 */

const makeSelectGradesForDistrict = () =>
  createSelector(
    selectForm,
    formState => formState && formState.getIn(['values', 'metaData', META_DATA_GRADES])
  );

const makeSelectFormErrors = () =>
  createSelector(selectForm, formState => formState && formState.getIn(['syncErrors']));

const makeSelectFormServerErrors = () =>
  createSelector(selectForm, formState => formState && formState.getIn(['values', 'serverErrors']));

const makeSelectSubmitErrors = () =>
  createSelector(selectForm, formState => formState && formState.get('submitErrors'));

const makeSelectAllFormMetaData = () =>
  createSelector(makeSelectGradesForDistrict(), grades => fromJS({ grades }));

const makeSelectSchoolYearDates = () => schoolFormState =>
  selectSchoolForm(schoolFormState, 'schoolYearStart', 'schoolYearEnd');

export {
  selectForm,
  makeSelectFormErrors,
  makeSelectAllFormMetaData,
  makeSelectFormServerErrors,
  makeSelectGradesForDistrict,
  makeSelectSchoolYearDates,
  makeSelectSubmitErrors,
};
