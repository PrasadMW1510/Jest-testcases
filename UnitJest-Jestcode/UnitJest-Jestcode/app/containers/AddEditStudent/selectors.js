import { createSelector } from 'reselect';
import { FIELD_SUBMIT_ERRORS } from 'components/StudentForm/constants';
import { FORM_STUDENT_PROFILE } from './constants';

const selectForm = state => state.getIn(['form', FORM_STUDENT_PROFILE]);

const makeSelectFormErrors = () =>
  createSelector(selectForm, formState => formState && formState.getIn(['syncErrors']));

const makeSelectFormServerErrors = () =>
  createSelector(selectForm, formState => formState && formState.get(FIELD_SUBMIT_ERRORS));

export { selectForm, makeSelectFormErrors, makeSelectFormServerErrors };
