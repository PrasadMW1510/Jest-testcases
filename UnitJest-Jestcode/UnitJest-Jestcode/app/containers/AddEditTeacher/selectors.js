import { createSelector } from 'reselect';
import { FORM_TEACHER_PROFILE } from './constants';

/**
 * Direct selector to the form state domain
 */
const selectForms = state => state.get('form');

const makeSelectFormErrors = () =>
  createSelector(
    selectForms,
    formState => formState && formState.getIn([FORM_TEACHER_PROFILE, 'syncErrors'])
  );

const makeSelectSubmitErrors = () =>
  createSelector(
    selectForms,
    formState => formState && formState.getIn([FORM_TEACHER_PROFILE, 'submitErrors'])
  );

/**
 * Direct selector to the addEditTeacherContainer state domain
 */
const selectAddEditTeacherContainer = state => state.get('addEditTeacherContainer');

const makeSelectAddEditTeacherContainer = () =>
  createSelector(selectAddEditTeacherContainer, substate => substate && substate.toJS());

/**
 * Selector to get profile details for Add Edit Teacher
 */
const makeSelectAddEditTeacherProfileDetails = () =>
  createSelector(
    selectAddEditTeacherContainer,
    substate => substate && substate.get('profileDetails')
  );

const makeSelectAddEditTeacherPermissionsData = () =>
  createSelector(
    selectAddEditTeacherContainer,
    substate =>
      substate && substate.get('permissionsData').map(permission => Number(permission.id[0]))
  );

const makeSelectAddEditTeacherSchoolsAndClassesData = () =>
  createSelector(
    selectAddEditTeacherContainer,
    substate => substate && substate.get('schoolsAndClassesDetails')
  );

const makeSelectAddEditTeacherPasswordConfigData = () =>
  createSelector(
    selectAddEditTeacherContainer,
    substate => substate && substate.get('passwordConfig')
  );

// export the below selectors
export {
  makeSelectFormErrors,
  makeSelectSubmitErrors,
  makeSelectAddEditTeacherProfileDetails,
  makeSelectAddEditTeacherPermissionsData,
  makeSelectAddEditTeacherSchoolsAndClassesData,
  makeSelectAddEditTeacherPasswordConfigData,
  makeSelectAddEditTeacherContainer,
};
