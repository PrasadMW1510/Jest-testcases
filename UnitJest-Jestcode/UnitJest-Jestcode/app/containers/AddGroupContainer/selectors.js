import { createSelector } from 'reselect';

/**
 * Direct selector to the addGroupContainer state domain
 */
const selectAddGroupContainerDomain = state => state.get('addGroupContainer');

/**
 * Other specific selectors
 */
const makeSelectClassInfoWithStudent = () =>
  createSelector(selectAddGroupContainerDomain, substate => substate.get('classesWithStudents'));

const makeSelectPostGroupId = () =>
  createSelector(selectAddGroupContainerDomain, substate => substate.getIn(['postGroupId', 0]));

const makeSelectPostGroupFailure = () =>
  createSelector(selectAddGroupContainerDomain, substate => substate.getIn(['errorPostGroupId']));

const makeSelectGroupInfo = () =>
  createSelector(selectAddGroupContainerDomain, substate =>
    substate.getIn(['getGroupInfoSuccess'])
  );

export {
  selectAddGroupContainerDomain,
  makeSelectClassInfoWithStudent,
  makeSelectPostGroupId,
  makeSelectPostGroupFailure,
  makeSelectGroupInfo,
};
