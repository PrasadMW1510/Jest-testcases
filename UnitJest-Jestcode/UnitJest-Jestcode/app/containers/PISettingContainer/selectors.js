import { createSelector } from 'reselect';

/**
 * Direct selector to the piSettingContainer state domain
 */
const selectPISettingContainerDomain = state => state.get('piSettingContainer');

/**
 * Other specific selectors
 */

/**
 * Default selector used by PISettingContainer
 */

const makeSelectPISettingContainer = () =>
  createSelector(selectPISettingContainerDomain, substate => substate);
export { makeSelectPISettingContainer };
