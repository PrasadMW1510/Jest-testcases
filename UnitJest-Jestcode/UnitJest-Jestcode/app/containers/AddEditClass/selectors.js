import { fromJS } from 'immutable';
import { createSelector } from 'reselect';
import { makeSelectProgramAvailableData } from 'containers/App/selectors';
import {
  FORM_CLASS_PROFILE,
  META_DATA_GRADES,
  META_DATA_STUDENTS,
  META_DATA_TEACHERS,
} from './constants';

const selectForm = state => state.getIn(['form', FORM_CLASS_PROFILE]);

const makeSelectFormErrors = () =>
  createSelector(selectForm, formState => formState && formState.getIn(['syncErrors']));

const makeSelectFormServerErrors = () =>
  createSelector(selectForm, formState => formState && formState.getIn(['values', 'serverErrors']));

const makeSelectFormMetaDataGrades = () =>
  createSelector(
    selectForm,
    formState => formState && formState.getIn(['values', 'metaData', META_DATA_GRADES])
  );

const makeSelectFormMetaDataTeachers = () =>
  createSelector(
    selectForm,
    formState => formState && formState.getIn(['values', 'metaData', META_DATA_TEACHERS])
  );

const makeSelectFormMetaDataStudents = () =>
  createSelector(
    selectForm,
    formState => formState && formState.getIn(['values', 'metaData', META_DATA_STUDENTS])
  );

const makeSelectFormMetaData = () =>
  createSelector(
    makeSelectProgramAvailableData(),
    makeSelectFormMetaDataGrades(),
    makeSelectFormMetaDataTeachers(),
    makeSelectFormMetaDataStudents(),
    (applications, grades, teachers, students) =>
      fromJS({ applications, grades, students, teachers })
  );

export {
  selectForm,
  makeSelectFormErrors,
  makeSelectFormMetaData,
  makeSelectFormMetaDataGrades,
  makeSelectFormMetaDataStudents,
  makeSelectFormMetaDataTeachers,
  makeSelectFormServerErrors,
};
