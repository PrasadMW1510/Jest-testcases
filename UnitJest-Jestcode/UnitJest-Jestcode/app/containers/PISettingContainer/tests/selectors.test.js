import { fromJS } from 'immutable';
import { makeSelectPISettingContainer } from '../selectors';

describe('selectPISettingContainerDomain', () => {
  let piSettingContainer = null;
  let mockState = null;

  beforeEach(() => {
    piSettingContainer = fromJS({
      loading: false,
      setting: {},
    });
    mockState = fromJS({
      piSettingContainer,
    });
  });

  it('should consistently return the makeSelectPISettingContainer app state', () => {
    expect(makeSelectPISettingContainer()(mockState)).toMatchSnapshot();
  });
});
