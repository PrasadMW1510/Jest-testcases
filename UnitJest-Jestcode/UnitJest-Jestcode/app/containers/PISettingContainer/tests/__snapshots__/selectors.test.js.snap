// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectPISettingContainerDomain should consistently return the makeSelectPISettingContainer app state 1`] = `
Immutable.Map {
  loading: false,
  setting: Immutable.Map {
  },
}
`;
